src/components/lib/FlowGrid/FlowGrid.jsx

```diff
@@ -198,19 +220,40 @@ export default class FlowGrid extends Component{
     this._addIfNeededAndSelect(newLocation, isRight ? 'LEFT' : 'RIGHT')
   }
 
+  onFillTargetMouseDown(location) {
+    window.recorder.recordNamedEvent("FlowGrid set fillRegion state")
+    this.setState({tracingFillRegion: true, fillRegion: {start: location}})
+  }
+
+  onCellMouseUp(location) {
+    if (!this.state.tracingFillRegion) { return }
+    window.recorder.recordNamedEvent("FlowGrid set fillRegion state")
+    this.props.onFillRegion(this.state.fillRegion)
+    this._handleEndRangeSelect(this.state.fillRegion.end)
+    this.setState({tracingFillRegion: false, fillRegion: {}})
+  }
+
+  // TODO(matthew): Look into necessity of 'inSelectedRegion' passed to cell below.
   _cell(location) {
-    const item = this.props.items.find(i => isAtLocation(i.location, location));
+    const item = this.props.items.find(i => isAtLocation(i.location, location))
+    const {selectedCell, selectedRegion} = this.props
+    const inSelectedCell = isAtLocation(selectedCell, location)
+    const selectedRegionNontrivial = selectedRegion.length === 2 && !isAtLocation(selectedRegion[0], selectedRegion[1])
+    const hasNonEmptyItem = !!item && !this.props.isItemEmpty(item.key)
+    const showFillToken = inSelectedCell && hasNonEmptyItem && !this.state.dragSelecting && !selectedRegionNontrivial
```

Comment:
This kind of seems like it should be its own function.  Realize that would require some more work though.